File: api/Ganymede.Api.Data/Rulesets/Ruleset.cs
using System;
using System.ComponentModel.DataAnnotations.Schema;

namespace Ganymede.Api.Data.Rulesets
{
    public class Ruleset
    {
        public int ID { get; set; }
        public string Name { get; set; }
        public string Abbrevation { get; set; }
        public DateTime ReleaseDate { get; set; }
        
        public int PublisherID { get; set; }
        [ForeignKey("PublisherID")]
        public Publisher Publisher { get; set; }
    }
}

File: src/app/campaign/models/ruleset.ts
import { Publisher } from "./publisher";

export class Ruleset {
	public id: number;
	public name: string;
	public abbreviation: string;
	public releaseDate: Date;
	public publisher: Publisher;

	constructor() {
		this.id = this.name = this.abbreviation = this.releaseDate = null;
		this.publisher = new Publisher();
	}
}
